exporter/datadogexporter/metadata/host.go

```diff
@@ -0,0 +1,64 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metadata
+
+import (
+	"runtime"
+
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/config"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/metadata/system"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/utils/cache"
+)
+
+// GetHost gets the hostname according to configuration.
+// It gets the configuration hostname and if
+// not available it relies on the OS hostname
+func GetHost(logger *zap.Logger, cfg *config.Config) *string {
+	if cacheVal, ok := cache.Get(cache.CanonicalHostnameKey); ok {
+		return cacheVal.(*string)
+	}
+
+	if err := validHostname(cfg.Hostname); err == nil {
+		cache.SetNoExpire(cache.CanonicalHostnameKey, &cfg.Hostname)
+		return &cfg.Hostname
+	} else if cfg.Hostname != "" {
+		logger.Error("Hostname set in configuration is invalid", zap.Error(err))
+	}
+
+	// Get system hostname
+	hostInfo := system.GetHostInfo(logger)
+	hostname := hostInfo.FQDN
+	if err := validHostname(hostInfo.FQDN); err != nil {
+		// FQDN is always empty on Windows
+		// so we don't report the validity there
+		if runtime.GOOS != "windows" {
```

Comment:
This was meant as a temporary solution in case we find a way to get the FQDN on Windows, but since that seems unlikely I am going to go with the `validFQDN` solution, thanks for pointing it out :)